model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1087,6 +1087,44 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for artifacts stored in a local directory.
+    // payload: ArtifactFilePayload.
+    FILE     = 0 [(beam_urn) = "beam:artifact:file:v1"];
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: raw data bytes.
+    EMBEDDED = 1 [(beam_urn) = "beam:artifact:embedded:v1"];
+    // A URN for artifacts described by HTTP links.
+    // payload: a string for an artifact HTTP URL
+    HTTP     = 2 [(beam_urn) = "beam:artifact:http:v1"];
+    // A URN for artifacts hosted on PYPI.
+    // artifact_id: a PYPI project name
+    // version_range: a PYPI compatible version string
+    // payload: None
+    PYPI     = 3 [(beam_urn) = "beam:artifact:pypi:v1"];
+    // A URN for artifacts hosted on Maven central.
+    // artifact_id: [maven group id]:[maven artifact id]
+    // version_range: a Maven compatible version string
+    // payload: None
+    MAVEN    = 4 [(beam_urn) = "beam:artifact:maven:v1"];
+  }
+}
+
+message ArtifactFilePayload {
+  // A path to an artifact file on a local system.
+  string local_path = 1;
+  // A generated staged name (no path).
+  string staged_name = 2;
+}
+
+message ArtifactInformation {
+  string urn = 1;
+  bytes payload = 2;
+  string artifact_id = 3;
+  string version_range = 4;
+}
```

Comment:
Thanks for a good catch. Added staged_name for embedded payload.

Reply:
But we'll need it for more than just embedded payload. And the name itself may not be enough to determine the role (do we try to install all .tar.gz files in Python? Or are some just data?)

Reply:
I see your point however I wonder what other options we have. It looks hard to describe all possible roles of a given file as a proto field. Maybe we could just delegate how to interpret a given file name to environments? (for instance, predefined file name for installable .tar.gz package for Python)

Reply:
Role would be an opaque string. (Well, likely a URN, but the runner wouldn't do anything about it.) It would be provided by, and consumed by, the SDK.

Reply:
Do we really believe we will have a mix and match for "source" descriptions and "destination" descriptions?

Reply:
Yes. The .tar.gz is one such example. Another example is that we treat the a manually-specified dev version of apache_beam in a special way (currently it *must* be named python_dataflow_sdk.tar); it would be better to not have such magic names. 

SDKs can choose to leave it blank, but I'd rather this be explicit than implicit in the name.